src/VssDatabase.cpp

```diff
@@ -826,198 +778,72 @@ jsoncons::json VssDatabase::getPathForSet(const string &path, jsoncons::json val
   return setValues;
 }
 
-void VssDatabase::checkSetPermission(WsChannel& channel, const string & path) {
-    // check if all the paths have write access.
-    bool haveAccess = accessValidator_->checkWriteAccess(channel, path);
-    if (!haveAccess) {
-       stringstream msg;
-       msg << "Path(s) in set request do not have write access or is invalid";
-       throw noPermissionException(msg.str());
-    }
-}
 
-// Method for setting values to signals.
-void VssDatabase::setSignal(WsChannel& channel,
-                            const string &path,
-                            jsoncons::json valueJson) {
-  if (path == "") {
-    string msg = "Path is empty while setting";
-    throw genException(msg);
-  }
-  checkSetPermission(channel, path);
+jsoncons::json  VssDatabase::setSignal(WsChannel& channel, const VSSPath &path, jsoncons::json &value, bool gen1_compat) {
+  jsoncons::json answer;
   
-  jsoncons::json setValues;
+  answer["path"] = gen1_compat? path.getVSSGen1Path() : path.getVSSPath();
+
+  if (!accessValidator_->checkWriteAccess(channel, path )) {
+      stringstream msg;
+      msg << "No write  access to " << path.getVSSPath();
+      throw noPermissionException(msg.str());
+  }
 
+  jsoncons::json res; 
   {
     std::lock_guard<std::mutex> lock_guard(rwMutex_);
-    try {
-        setValues = getPathForSet(path, valueJson);
-    }
-    catch( noPathFoundonTree& e) {
-      throw e;
-    }
-    catch ( genException &e) {
-      logger_->Log(LogLevel::ERROR, "Exception VssDatabase::setSignal: " + string(e.what()));
-      throw e;
+    res = jsonpath::json_query(data_tree__, path.getJSONPath());
+    if (res.is_array() && res.size() == 1) {
+    jsoncons::json resJson = res[0];
+      if (resJson.contains("datatype")) {
+        string value_type = resJson["datatype"].as<string>();
+        checkTypeAndBound(logger_, value_type, value);
+        resJson.insert_or_assign("value", value);
+        {
+          jsonpath::json_replace(data_tree__, path.getJSONPath(), resJson);
+        }
+      }
+      else {
+        throw genException(path.getVSSPath()+ "is invalid for set"); //Todo better error message. (Does not propagate);
+      }
     }
   }
-  if (setValues.is_array()) {
-    HandleSet(setValues);
-  } else {
-    string msg = "Exception occured while setting data for " + path;
-    throw genException(msg);
-  }
+
+  return answer;
 }
 
+//Only needed for DBUS currently- 
 // Method for setting values to signals.
-void VssDatabase::setSignal(const string &path,
-                            jsoncons::json valueJson) {
-  if (path == "") {
+void VssDatabase::setSignal(const string &dbuspath,
```

Comment:
path here is really dbus specific? Comment is good, but change the name here is wired.

Reply:
This function is only used in DBUS code path. Options where removing it,with complete Dbus support, leaving it for the time being, or upgrading DBUS code to reuse websocket code paths (which may be useful one day, but currrently I do not know, what that code is doing, and if it is really functional